compiler/optimizer/optimizer.go

```diff
@@ -132,6 +132,7 @@ func (o *Optimizer) Optimize(seq dag.Seq) (dag.Seq, error) {
 	seq = removePassOps(seq)
 	o.optimizeParallels(seq)
 	seq = mergeFilters(seq)
+	seq = insertDemandTests(seq)
```

Comment:
@nwt It would be nice to hide this behind some cli flag if it's not too fiddly. Otherwise I'll just take it out before merging.

Reply:
Since this is just temporary, how about an environment variable, which requires a little less plumbing?
```suggestion
	if os.Getenv("ZED_DEMAND_TESTS") != "" {
		seq = insertDemandTests(seq)
	}
```